servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/MultiAddressUrlHttpClientTest.java

```diff
@@ -253,26 +257,52 @@ void requestWithAsteriskFormRequestTargetWithHostHeader() {
         assertThat(subscriber.awaitOnError(), is(instanceOf(MalformedURLException.class)));
     }
 
+    @Test
+    void exactRequestAuthorityIsRepresentedInHostHeader() throws Exception {
+        try (StreamingHttpClient client = HttpClients
+                .forMultiAddressUrl(ID).defaultHttpPort(serverPort).buildStreaming()) {
+            // Request without a port in the authority
+            StreamingHttpRequest request = client.get(format("http://%s/200", serverHost));
+            // the host header should exactly match the authority provided.
+            requestAndValidate(client, request, OK, "/200", serverHost);
+
+            // Request with a port in the authority
+            request = client.get(format("http://%s:%d/200", serverHost, serverPort));
+            // the host header should exactly match the authority provided.
+            requestAndValidate(client, request, OK, "/200", serverHost + ":" + serverPort);
+
+            // Excludes userinfo and the '@'
+            request = client.get(format("http://user:pass@%s/200", serverHost, serverPort));
```

Comment:
here you are passing the port to format, but no `%d`?

Reply:
Good catch.